### PR TITLE
Small Ability Effect Move Block refactor

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Refactor `ABILITYEFFECT_MOVES_BLOCK` for better readability. Also fixes a bug with Dazzling abilities in doubles where the partner was not protected.

### CODE REVIEW

File: src/battle_util.c [hunk 1]
```diff
@@ -4977,46 +4977,76 @@ u32 AbilityBattleEffects(u32 caseID, u32 battler, u32 ability, u32 special, u32
...
-            u16 battlerAbility = GetBattlerAbility(battler);
-            u16 targetAbility = GetBattlerAbility(gBattlerTarget);
 
-            if ((gLastUsedAbility == ABILITY_SOUNDPROOF && gMovesInfo[move].soundMove && !(moveTarget & MOVE_TARGET_USER))
-             || (gLastUsedAbility == ABILITY_BULLETPROOF && gMovesInfo[move].ballisticMove))
+            switch (gLastUsedAbility)
+            {
+            case ABILITY_SOUNDPROOF:
+                if (gMovesInfo[move].soundMove && !(moveTarget & MOVE_TARGET_USER))
+                    effect = 1;
```
Comment (Bassoonian) [inline src/battle_util.c:4985]: Should/can we get an enum to describe what precisely these effect numbers stand for? It's not very intuitive

File: test/battle/ability/armor_tail.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#include "global.h"
+#include "test/battle.h"
+
+
+ASSUMPTIONS
+{
+    ASSUME(gMovesInfo[MOVE_QUICK_ATTACK].priority > 0);
+}
+
+DOUBLE_BATTLE_TEST("Armor Tail protects user from priority moves")
```
Comment (Bassoonian) [inline test/battle/ability/armor_tail.c:10]: I think we need @AsparagusEduardo's opinion, but do we really need four files with the majority of the code copy-pasted? Can't we just have the tests in common parametrized and add references to "tests for this ability are in file X", as is already the case in other test files I believe?

File: src/battle_util.c (hunk 1 above)
Comment (AlexOn1ine) [reply src/battle_util.c:4985]: Probably not feasible. They are used all across `AbilityBattleEffects` and can mean anything.

Comment (Bassoonian) [reply src/battle_util.c:4985]: Then I think we should streamline it. Out of scope for this PR of course, but it just makes it unnecessarily opaque to see what actually happens

File: test/battle/ability/armor_tail.c (hunk 2 above)
Comment (AsparagusEduardo) [reply test/battle/ability/armor_tail.c:10]: > Can't we just have the tests in common parametrized and add references to "tests for this ability are in file X", as is already the case in other test files I believe?

Alright.

Comment (AlexOn1ine) [reply test/battle/ability/armor_tail.c]: combined them

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**